src/InsertionsClient/Api/Props/Models/SwrFile.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Net.Insertions.Api.Models
+{
+    /// <summary>
+    /// Represents an SWR file in a VS repo local working directory.
+    /// </summary>
+    internal class SwrFile
+    {
+        /// <summary>
+        /// Full path to the swr file.
+        /// </summary>
+        public readonly string Path;
+
+        /// <summary>
+        /// Paths to payloads mentioned in the file.
+        /// </summary>
+        public readonly List<PayloadPath> PayloadPaths = new List<PayloadPath>();
+
+        /// <summary>
+        /// Represents a payload defined in the swr file.
+        /// </summary>
+        public class PayloadPath
```

Comment:
It doesn't have to be. I put it here, because `PayloadPath` type is only for representing the payloads _inside swr files_. So the type is quite closely related to the containing class. Do you think it shouldn't be nested?